src/dns_internal.cc

```diff
@@ -833,7 +833,7 @@ idnsDoSendQueryVC(nsvc *vc)
     vc->busy = 1;
 
     // Comm needs seconds but idnsCheckQueue() will check the exact timeout
-    const int timeout = (Config.Timeout.idns_query % 1000 ?
+    const time_t timeout = (Config.Timeout.idns_query % 1000 ?
                          Config.Timeout.idns_query + 1000 : Config.Timeout.idns_query) / 1000;
```

Comment:
If you are fixing this caller, please also fix a very similar `int` caller in src/helper.cc near line 1603:

```C++
const int timeLeft = max(1, (static_cast<int>(srv->parent->timeout) - timeSpent));
```

The explicit cast here is probably meant to make the max() template compile. There are several ways to adjust that, including something like that:

```C++
const time_t minimumNewTimeout = 1; // second
const auto timeLeft = max(minimumNewTimeout, srv->parent->timeout - timeSpent);
```

Reply:
Thanks for catching this, it had slipped past me. Fixing.

Reply:
> I'd rather static_cast<time_t>(1) ; is there any reason not to?

I prefer the two-variable suggestion for two reasons:

* It is arguably a lot more readable.
* It documents the effect of having that `1` in std::max().

I do not insist on my variation.